extensions/src/main/java/org/pentaho/platform/web/http/api/resources/utils/FileUtils.java

```diff
@@ -93,26 +94,39 @@ public static boolean containsReservedCharacter( String path, char[] reserved )
   }
 
   public static String getParentPath( final String path ) {
-    if ( path == null ) {
+    if ( path == null || path.length() == 0 || path.indexOf( '/' ) == -1 ) {
       throw new IllegalArgumentException();
-    } else if ( PATH_SEPARATOR.equals( path ) ) {
+    } else if ( path.equals( PATH_SEPARATOR ) ) {
       return null;
     }
-    int lastSlashIndex = path.lastIndexOf( PATH_SEPARATOR );
-    if ( lastSlashIndex == 0 ) {
+
+    int index = trimTrailingSlash( path ).lastIndexOf( PATH_SEPARATOR );
+
+    if(index == -1) {
+      return null;
+    } else if(index == 0) {
       return PATH_SEPARATOR;
-    } else if ( lastSlashIndex > 0 ) {
-      return path.substring( 0, lastSlashIndex );
     } else {
-      throw new IllegalArgumentException();
+      return path.substring( 0, index );
     }
   }
 
-  public static boolean isRootPath( final String path ) {
-    return PATH_SEPARATOR.equals( path );
+  public static boolean isRootLevelPath( final String path ) {
+    if ( path == null || path.length() == 0 || !path.startsWith( PATH_SEPARATOR ) ) {
+      // the path must be an absolute path like stated in the documentation for the REST API
+      throw new IllegalArgumentException();
+    } else if ( path.equals( PATH_SEPARATOR ) ) {
+      return true;
+    } else if ( trimTrailingSlash( path ).startsWith( ClientRepositoryPaths.getPublicFolderPath() + PATH_SEPARATOR )
+      || trimTrailingSlash( path ).startsWith( ClientRepositoryPaths.getHomeFolderPath() + PATH_SEPARATOR )
+      || trimTrailingSlash( path ).startsWith( ClientRepositoryPaths.getEtcFolderPath() + PATH_SEPARATOR ) ) {
+      return false;
+    }
+
+    return true;
   }
 
-  public static boolean isRootLevelPath( final String path ) {
-    return ( path != null ) && !isRootPath( path ) && path.startsWith( PATH_SEPARATOR ) && isRootPath( getParentPath( path ) );
+  private static String trimTrailingSlash( String path ) {
+    return path.endsWith( PATH_SEPARATOR ) ? path.substring( 0, path.length() - 2 ) : path;
```

Comment:
Null check was removed . . . why ?

Looking at the entire file, the method is private, and null is prevented before calling the method.